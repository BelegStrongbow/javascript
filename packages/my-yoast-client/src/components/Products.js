import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import Product from "./Product";
import { ListHeading } from "./Headings";

const ProductsContainer = styled.div`
	display: flex;
	background-color: ${ colors.$color_white };
	padding: 0 24px 24px 24px;
	margin: 0 0 0 8px;
	flex-wrap: wrap;
	width: 100%;
	align-items: stretch;
`;

const ProductsListHeading = styled( ListHeading )`
	width: calc( 100% - 16px );
	font-size: 22px;
	margin: 0 0 24px 0px;
	padding: 0 0 24px 0;
`;

/**
 * Returns the rendered Products component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered products page.
 */
export default class Products extends React.Component {
	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		const props = this.props;
		return (
			<ProductsContainer>
				<ProductsListHeading>
					{ props.heading }
					{ props.byLine }
				</ProductsListHeading>
				{ props.noResults }
				{ props.products.map( function( product ) {
					return <Product
						key={ product.glNumber }
						name={ product.name }
						glNumber={ product.glNumber }
						currentVersion={ product.currentVersion }
						icon={ product.icon }
						buttons={ product.buttons }
						composerToken={ props.composerToken }
						onComposerHelpModalOpen={ props.onComposerHelpModalOpen }
						onComposerHelpModalClose={ props.onComposerHelpModalClose }
						type={ product.category }
					/>;
				} ) }
			</ProductsContainer>
		);
	}
}

Products.propTypes = {
	byLine: PropTypes.element,
	heading: PropTypes.string.isRequired,
	noResults: PropTypes.string,
	composerToken: PropTypes.object,
	onComposerHelpModalOpen: PropTypes.func,
	onComposerHelpModalClose: PropTypes.func,
	composerHelpProductName: PropTypes.string,
};

Products.defaultProps = {
	byLine: null,
	noResults: "",
	onComposerHelpModalClose: () => {},
	onComposerHelpModalOpen: () => {},
};
