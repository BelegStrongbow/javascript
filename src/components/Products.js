import React from "react";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import Product from "./Product";
import { ListHeading } from "./ListHeading";

const ProductsContainer = styled.div`
	display: flex;
	background-color: ${ colors.$color_white };
	padding: 0 24px 24px 24px;
	margin: 0 0 0 8px;
	flex-wrap: wrap;
	width: 100%;
	align-items: flex-start;
`;

const ProductsListHeading = styled( ListHeading )`
	width: calc( 100% - 16px );
	font-size: 22px;
	margin: 0 0 24px 0px;
	padding: 0 0 24px 0;
`;

/**
 * Returns the rendered Products component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered products page.
 */
export default class Products extends React.Component {
	render() {
		let props = this.props;
		return (
			<ProductsContainer>
				<ProductsListHeading>
					{ props.heading }
					{ props.byLine }
				</ProductsListHeading>
					{ props.noResults }
					{ props.products.map( function( product ) {
						return <Product
							key={ product.id }
							name={ product.name }
							currentVersion={ product.currentVersion }
							icon={ product.icon }
							buttons={ product.buttons }
						/>;
					} ) }
			</ProductsContainer>
		);
	}
}

Products.propTypes = {
	byLine: React.PropTypes.element,
	heading: React.PropTypes.string.isRequired,
	noResults: React.PropTypes.string,
};

Products.defaultProps = {
	byLine: null,
	noResults: "",
};
