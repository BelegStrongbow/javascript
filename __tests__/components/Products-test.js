import React from 'react';
import { createComponentWithIntl } from "../../utils";
import Products from '../../src/components/Products';
import SeoIcon from "../../src/icons/Yoast/Yoast_SEO_Icon_Small.svg";

test( 'The Products component matches the snapshot', () => {
	let products = [
		{
			id: "1",
			name: "Yoast SEO",
			currentVersion: "4.7",
			icon: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
				{ label: "for Drupal",
					onButtonClick: () => {} },
			],
		}, {
			id: "2",
			name: "Local SEO",
			currentVersion: "4.7",
			icon: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
			],
		}, {
			id: "3",
			name: "Video SEO",
			currentVersion: "4.7",
			icon: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
				{ label: "for Drupal",
					onButtonClick: () => {} },
				{ label: "for Typo3",
					onButtonClick: () => {} },
			],
		}, {
			id: "4",
			name: "News SEO",
			currentVersion: "4.7",
			icon: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
				{ label: "for Drupal",
					onButtonClick: () => {} },
				{ label: "for Typo3",
					onButtonClick: () => {} },
			],
		}, {
			id: "5",
			name: "Yoast SEO for WooCommerce",
			currentVersion: "4.7",
			icon: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
			],
		},
	];

	const component = createComponentWithIntl(
		<Products products={ products } heading="Products" byLine={ <span> - Buy here!</span> }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
});
