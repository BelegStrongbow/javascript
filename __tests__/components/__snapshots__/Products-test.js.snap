exports[`test The Products component matches the snapshot 1`] = `
<div
  className="jhQvCy">
  <h2
    className="iJNKGZ">
    Products
    <span>
       - Buy here!
    </span>
  </h2>
  
  <div
    className="cbzkol">
    <h3
      className="eHeqpp">
      Yoast SEO
    </h3>
    <p
      className="jPEgOF">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="boOFOz">
      <li
        className="jVUpbY">
        <button
          aria-label="Download Yoast SEO for Wordpress"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Wordpress
        </p>
      </li>
      <li
        className="jVUpbY">
        <button
          aria-label="Download Yoast SEO for Drupal"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Drupal
        </p>
      </li>
    </ul>
  </div>
  <div
    className="cbzkol">
    <h3
      className="eHeqpp">
      Local SEO
    </h3>
    <p
      className="jPEgOF">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="boOFOz">
      <li
        className="jVUpbY">
        <button
          aria-label="Download Local SEO for Wordpress"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Wordpress
        </p>
      </li>
    </ul>
  </div>
  <div
    className="cbzkol">
    <h3
      className="eHeqpp">
      Video SEO
    </h3>
    <p
      className="jPEgOF">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="boOFOz">
      <li
        className="jVUpbY">
        <button
          aria-label="Download Video SEO for Wordpress"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Wordpress
        </p>
      </li>
      <li
        className="jVUpbY">
        <button
          aria-label="Download Video SEO for Drupal"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Drupal
        </p>
      </li>
      <li
        className="jVUpbY">
        <button
          aria-label="Download Video SEO for Typo3"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Typo3
        </p>
      </li>
    </ul>
  </div>
  <div
    className="cbzkol">
    <h3
      className="eHeqpp">
      News SEO
    </h3>
    <p
      className="jPEgOF">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="boOFOz">
      <li
        className="jVUpbY">
        <button
          aria-label="Download News SEO for Wordpress"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Wordpress
        </p>
      </li>
      <li
        className="jVUpbY">
        <button
          aria-label="Download News SEO for Drupal"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Drupal
        </p>
      </li>
      <li
        className="jVUpbY">
        <button
          aria-label="Download News SEO for Typo3"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Typo3
        </p>
      </li>
    </ul>
  </div>
  <div
    className="cbzkol">
    <h3
      className="eHeqpp">
      Yoast SEO for WooCommerce
    </h3>
    <p
      className="jPEgOF">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="boOFOz">
      <li
        className="jVUpbY">
        <button
          aria-label="Download Yoast SEO for WooCommerce for Wordpress"
          className="bSTGwY"
          onClick={[Function]}>
          Download
        </button>
        <p
          aria-hidden="true"
          className="cBxjGK">
          for Wordpress
        </p>
      </li>
    </ul>
  </div>
</div>
`;
